satellite/console/consoleweb/server.go

```diff
@@ -192,6 +217,323 @@ func (server *Server) appHandler(w http.ResponseWriter, r *http.Request) {
+	if err != nil {
+		server.serveJsonError(w, 401, err)
+		return
+	}
+
+	type ChangePasswordRequestModel struct {
+		Password    string `json:"password"`
+		NewPassword string `json:"newPassword"`
+	}
+
+	var passwordChange ChangePasswordRequestModel
+	err = json.NewDecoder(r.Body).Decode(&passwordChange)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	err = server.service.ChangePassword(ctx, passwordChange.Password, passwordChange.NewPassword)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		return
+	}
+}
+
+// createNewUserRequestHandler gets password hash value and creates new inactive User
+func (server *Server) createNewUserRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type createUserRequestModel struct {
+		console.CreateUser
+		Secret         string `json:"secret"`
+		ReferrerUserID string `json:"referrerUserId"`
+	}
+
+	var model createUserRequestModel
+	err = json.NewDecoder(r.Body).Decode(&model)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	secret, err := console.RegistrationSecretFromBase64(model.Secret)
+	if err != nil {
+		log.Error("register: failed to create account",
+			zap.Error(err))
+		log.Debug("register: ", zap.String("rawSecret", model.Secret))
+		server.serveJsonError(w, 400, err)
+
+		return
+	}
+
+	user, err := server.service.CreateUser(ctx, model.CreateUser, secret, model.ReferrerUserID)
+	if err != nil {
+		log.Error("register: failed to create account",
+			zap.Error(err))
+		log.Debug("register: ", zap.String("rawSecret", model.Secret))
+		server.serveJsonError(w, 400, err)
+
+		return
+	}
+
+	token, err := server.service.GenerateActivationToken(ctx, user.ID, user.Email)
+	if err != nil {
+		log.Error("register: failed to generate activation token",
+			zap.Stringer("id", user.ID),
+			zap.String("email", user.Email),
+			zap.Error(err))
+		server.serveJsonError(w, 400, err)
+
+		return
+	}
+
+	rootObject, ok := ctx.Value("rootObject").(RootObject)
+	if !ok {
+		server.log.Error("root object is not set")
+		return
+	}
+	link := rootObject.Origin + rootObject.ActivationPath + token
+	userName := user.ShortName
+	if user.ShortName == "" {
+		userName = user.FullName
+	}
+
+	server.mailService.SendRenderedAsync(
+		ctx,
+		[]post.Address{{Address: user.Email, Name: userName}},
+		&consoleql.AccountActivationEmail{
+			Origin:         rootObject.Origin,
+			ActivationLink: link,
+		},
+	)
+
+	err = json.NewEncoder(w).Encode(user)
+	if err != nil {
+		server.serveJsonError(w, 500, err)
+		server.log.Debug("Error serializing response: " + err.Error())
+	}
+}
+
+// deleteAccountRequestHandler deletes User
+func (server *Server) deleteAccountRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	auth, err := console.GetAuth(ctx)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		w.WriteHeader(401)
```

Comment:
Why do we need this extra `w.WriteHeader`? you can not set header after `w.Write` or `w.WriteHeader` has been called